vm/src/execute.rs

```diff
@@ -68,7 +68,21 @@ pub fn execute(
 ) -> Result<CallReceipt<Result<Vec<u8>, ContractError>>, Error> {
     // Transaction will be discarded if it is a deployment transaction
     // with gas limit smaller than deploy charge.
-    deploy_check(tx, config)?;
+    tx.deploy_check(
+        config.gas_per_deploy_byte,
+        config.min_deploy_gas_price,
+        config.min_deploy_points,
+    )
+    .map_err(|e| Error::Panic(e.legacy_to_string()))?;
+    let blob_min_charge = tx
+        .blob_check(config.gas_per_blob)
+        .map_err(|e| Error::Panic(e.legacy_to_string()))?;
+
+    if blob_min_charge.is_some() && !config.blob_enable {
```

Comment:
Maybe we can make this check a bit more comprehensible...
I suggest to make it panic if `self.blob().is_some()` and move the check inside `blob_check`.
So we can also avoid having `min_charge` as an Option

Reply:
Additionally, being an option allow us to know if that's a blob tx after being stripped, without relying on min_charge>0

Reply:
> Additionally, being an option allow us to know if that's a blob tx after being stripped, without relying on min_charge>0

You are relying on min_charge right now (you check if it's Some)... which is precisely what I dislike.
You can easily check if it's a blob tx by using the original tx (the stripped one is a clone)

Reply:
No, the stripped one is converted to memory